tests/dragonfly/snapshot_test.py

```diff
@@ -500,3 +502,33 @@ async def test_tiered_entries_throttle(async_client: aioredis.Redis):
 
     await load_task
     assert await StaticSeeder.capture(async_client) == start_capture
+
+
+@dfly_args({"proactor_threads": 1, "dbfilename": "big_snapshot"})
+async def test_big_value_serialization_memory_limit(df_factory):
+    instance = df_factory.create()
+    instance.start()
+    client = instance.client()
+
+    ten_mb = 125000
+
+    def ten_mb_random_string():
+        return "".join(random.choices(string.ascii_letters, k=ten_mb))
+
+    one_gb = 125000000
+    upper_limit = one_gb * 1.25
+
+    while instance.rss < one_gb:
+        await client.execute_command(f"SADD foo {ten_mb_random_string()}")
+
+    async def check_memory_usage(instance):
+        while True:
+            assert instance.rss < upper_limit
```

Comment:
what is this `rss` field? where is the value gathered and how frequently does it update?

Reply:
`.rss` is a `function` but marked as `property`. It's updated everytime on the loop `instance.rss` is actually syntactic sugar for `instance.rss()`